packages/ripple-binary-codec/test/shamap.test.js

```diff
@@ -28,7 +31,19 @@ function makeItem(indexArg) {
 }
 
 describe('ShaMap', () => {
-  now()
+  test('you can add prehashed items to trees', () => {
+    const fixture = loadFixture('ledger-full-40000.json')
+    const fullMap = new ShaMap()
+    fixture.accountState.forEach((entry) => {
+      const [index, item] = entryItemizer(entry)
+      fullMap.addItem(index, item)
+    })
+    const shortMap = new ShaMap()
+    fullMap.walkLeaves((leaf) => {
+      shortMap.addItem(leaf.index, { preHashed: leaf.hash() })
+    })
+    expect(fullMap.hash().toHex()).toBe(shortMap.hash().toHex())
```

Comment:
https://github.com/Xahau/Validation-Ledger-Tx-Store-to-xPOP/blob/main/lib/xpop/v1.mjs#L301